lxd/db/generate/README.md

```diff
@@ -0,0 +1,187 @@
+# `lxd-generate`
+
+## Introduction
+
+`lxd-generate` is a database statement and associated `go` function generator
+for LXD and related projects. `lxd-generate` utilizes `go`'s code generation
+directives (`//go:generate ...`) alongside go's [ast](https://pkg.go.dev/go/ast)
+package for parsing the syntax tree for go structs and variables. We use
+`lxd-generate` for the majority of our SQL statements and database interactions
+on the `go` side for consistency and predictability.
+
+## Usage
+
+### Initialization
+
+Generally the first thing we will want to do for any newly generated file is to
+establish the command, the target file, and ensure the file has been cleared of
+content:
+
+```go
+//go:generate -command mapper lxd-generate db mapper -t instances.mapper.go
+//go:generate mapper reset -i -b "//go:build linux && cgo && !agent"
+
+```
+
+This will initiate a call to `lxd-generate db mapper -t instances.mapper.go` on
+each call to the go generation directive `mapper`.
+
+### SQL Statement generation
+
+SQL generation supports the following SQL statement types:
+Type                                  | Description
+:---                                  | :----
+`objects`                             | Creates a basic SELECT statement of the form `SELECT <columns> FROM <table> ORDER BY <columns>`.
+`objects-by-<FIELD>-and-<FIELD>...`   | Parses a pre-existing SELECT statement of the form produced by`objects`, and appends a `WHERE` clause with the given fields located in the associated struct.
```

Comment:
So it should be exactly of the form produced by the `objects` directive, so that is 
```go
var <entity>Objects = <database>.RegisterStmt(`SQL String does here`)
```

I've added this to the description as well now.